### PR TITLE
[dashboard] Unify and standardize InfoBox and AlertBox components

### PR DESCRIPTION
**Pull request by jankeromnes**
Fixes https://github.com/gitpod-io/gitpod/issues/3996

TODO:
- [x] InfoBox component
- [x] AlertBox component (but not all alerts were unified, see https://github.com/gitpod-io/gitpod/pull/4020#issuecomment-823847403)

### CODE REVIEW

File: components/dashboard/src/components/InfoBox.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import info from '../images/info.svg';
+
+export default function InfoBox(p: { className?: string, children?: React.ReactNode }) {
+    return <div className={'flex rounded-md bg-gray-200 dark:bg-gray-800 text-gray-600 dark:text-gray-400 p-4 ' + (p.className || '')}>
+        <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+        {p.children}
```
Comment (jankeromnes) [inline components/dashboard/src/components/InfoBox.tsx:12]: Screenshot 2021-04-20 at 18 52 47

```suggestion
        <span>{p.children}</span>
```

**Comment by jankeromnes** [comment]
IO preview: https://jx-cancel-downgrade.staging.gitpod-io-dev.com/workspaces

Unified InfoBoxes:

Screenshot 2021-04-20 at 19 26 20
Screenshot 2021-04-20 at 19 26 30
Screenshot 2021-04-20 at 19 27 27
Screenshot 2021-04-20 at 19 26 42
Screenshot 2021-04-20 at 19 26 52
Screenshot 2021-04-20 at 19 27 15

**Comment by jankeromnes** [comment]
Unified AlertBoxes:

Screenshot 2021-04-21 at 09 30 16
Screenshot 2021-04-21 at 09 28 06
Screenshot 2021-04-21 at 09 29 55
Screenshot 2021-04-21 at 09 29 00
Screenshot 2021-04-21 at 09 33 02
Screenshot 2021-04-21 at 09 33 13

And also:

https://github.com/gitpod-io/gitpod/blob/169b7e10b8099bbec9eb42d1e75825156dcfebdd/components/dashboard/src/settings/Integrations.tsx#L588

Not yet unified "AlertBoxes" (should we unify these too?):

Screenshot 2021-04-21 at 09 34 11
Screenshot 2021-04-21 at 09 34 01

**Comment by jankeromnes** [comment]
@gtsiolis Could you please take a look when you have time?

Priority: 🔅 Low.

**Review by gtsiolis — APPROVED** [review]
Thanks for componetizing this element @jankeromnes! 💯  

Added a few more unrelated comments to colors. We're still missing some color correction in modals but I think we can tackle these separately once we componetize the modals.

**Comment by gtsiolis** [comment]
Also, thank you for adding all these screenshots! These are really helpful during reviewing visual changes.

**Tip**: I usually find the table element in markdown extremely useful to represent such information. For example, using for BEFORE / AFTER screenshots or adding some structure to unrelated screenshots. 💡 

```
# EXAMPLE A

| BEFORE | AFTER |
|-|-|
| ![image](...) | ![image](...) | 

# EXAMPLE B

| Screenshot A | Screenshot B | Screenshot C |
|-|-|-|
| ![image](...) | ![image](...) | ![image](...) | 
```

File: components/dashboard/src/settings/Plans.tsx [hunk 2]
```diff
@@ -448,22 +448,19 @@ export default function () {
...
-                <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
-                <span>If you are interested in purchasing a plan for a team, purchase a Team plan with one centralized billing. <a className="underline" href="https://www.gitpod.io/docs/teams/" target="_blank">Learn more</a></span>
-            </div>
+            <InfoBox className="w-2/3 mt-14 mx-auto">
+                If you are interested in purchasing a plan for a team, purchase a Team plan with one centralized billing. <a className="underline" href="https://www.gitpod.io/docs/teams/">Learn more</a>
+            </InfoBox>
             {!!confirmUpgradeToPlan && <Modal visible={true} onClose={() => setConfirmUpgradeToPlan(undefined)}>
                 <h3>Upgrade to {confirmUpgradeToPlan.name}</h3>
                 <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-4 -mx-6 px-6 py-2">
                     <p className="mt-1 mb-4 text-base">You are about to upgrade to {confirmUpgradeToPlan.name}.</p>
```
[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/dashboard/src/settings/Plans.tsx:457]: Good catch! Many other modals already have `text-gray-500` for both light and dark. Will align here, but leave additional componentization of Modals as follow-up.

File: components/dashboard/src/components/AlertBox.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import exclamation from '../images/exclamation.svg';
+
+export default function InfoBox(p: { className?: string, children?: React.ReactNode }) {
+    return <div className={'flex rounded-md bg-gitpod-kumquat-light text-gitpod-red p-4 ' + (p.className || '')}>
```
[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/dashboard/src/components/AlertBox.tsx:10]: Sure! Let's try `rounded-xl` for both `AlertBox` and `InfoBox`. 🚀

**Comment by jankeromnes** [comment]
Many thanks for the great review! 💯

> Also, thank you for adding all these screenshots! These are really helpful during reviewing visual changes.

Sure! Glad they're helpful. ✨

> **Tip**: I usually find the table element in markdown extremely useful to represent such information. For example, using for BEFORE / AFTER screenshots or adding some structure to unrelated screenshots. Feel free to try this or keep adding your awesome contributions without any table!💡 

Haha, great tip! I [agree](https://github.com/gitpod-io/gitpod/pull/3901#issuecomment-818528370), although I typically go old-school with a bunch of:

```html
<table><tr><td>
<img/>
</td><td>
<img/>
</td></tr></table>
```

(😅)

**Comment by jankeromnes** [comment]
Also, what should we do about the "Not yet unified "AlertBoxes" in https://github.com/gitpod-io/gitpod/pull/4020#issuecomment-823847403 ?

They currently have a different visual style, which is why I haven't unified them yet. Should they stay different for now?

**Comment by gtsiolis** [comment]
Go `<table>`!

> They currently have a different visual style ...

Yes, the ones within the modals should eventually use the same visual style. The garbage collection alert could use a larger component variant.

**Pull request merged**